accounts, atm, auth, branch, profiles, statements: cleanup

mgfcmartinez opened the pull request:
- Fix typos and grammatical errors, try to make the writing style more
  consistent

- Remove BasePath from swagger definitions to prevent it being doubled
  (e.g. /v1/v1/accounts/check)

- Fix missing/incorrect response schemas

- Make godoc comments more consistent

- Fix endpoint tags to ensure correct grouping

mgfcmartinez commented:
@nkristianto @ghazninattarshah to facilitate easier reviewing, i'll include the generated files once the changes to the actual definitions have been approved

ghazninattarshah commented on v1/accounts/accounts.gunk:
```diff
@@ -517,9 +514,9 @@ type AccountService interface {
-	//         Summary:     "Update the status of the account",
+	//         Tags:        []string{"Account"},
+	//         Description: "Updates the status of an account.",
+	//         Summary:     "Update an account's status",
```
You shall remove `'s`, just say `account`

ghazninattarshah commented on v1/accounts/accounts.gunk:
```diff
@@ -539,22 +536,22 @@ type AccountService interface {
-	//         Summary:     "Check whether account exists or not",
+	//         Tags:        []string{"Account"},
+	//         Description: "Verify whether or not an account exists.",
+	//         Summary:     "Verify an account's existence",
```
You shall remove `'s`, just say `account`

ghazninattarshah commented on v1/atm/atm.gunk:
```diff
@@ -89,55 +87,60 @@ type CreateATMRequest struct {
 	//                         Description: "Request executed successfully.",
+	//                         Schema: openapiv2.Schema{
+	//                                 JSONSchema: openapiv2.JSONSchema{
+	//                                         Ref: "#/definitions/atmATM",
```
Should the Ref be "#/definitions/atm`s`ATM"?

mgfcmartinez replied:
no, since the package name is singular: https://github.com/openbank/openbank/blob/b496bf680a51b141c09f9cb480b133d4d54aba6e/v1/atm/all.swagger.json#L173

ghazninattarshah replied:
the gunk http path on each function starts with `/atms/...`? if it's meant to be singular, should that be `/atm/...`?

mgfcmartinez replied:
i understand your concern, i do think the `atm` package might need to be renamed for consistency with the other packages (which are plural), but any non-documentation changes are out of scope for this PR. it can be addressed in a future PR

ghazninattarshah commented on v1/atm/atm.gunk:
```diff
@@ -157,19 +160,24 @@ type ATMService interface {
 	//                         Description: "Request executed successfully.",
+	//                         Schema: openapiv2.Schema{
+	//                                 JSONSchema: openapiv2.JSONSchema{
+	//                                         Ref: "#/definitions/atmGetATMsResponse",
```
Should the Ref be "#/definitions/atm`s`GetATMsResponse"?

mgfcmartinez replied:
same as https://github.com/openbank/openbank/pull/17#discussion_r314579305

ghazninattarshah commented on v1/atm/atm.gunk:
```diff
@@ -203,6 +211,11 @@ type ATMService interface {
 	//                         Description: "Transaction created successfully.",
+	//                         Schema: openapiv2.Schema{
+	//                                 JSONSchema: openapiv2.JSONSchema{
+	//                                         Ref: "#/definitions/atmCreateATMResponse",
```
Should the Ref be "#/definitions/atm`s`CreateATMResponse"?

mgfcmartinez replied:
https://github.com/openbank/openbank/pull/17#discussion_r314579305

ghazninattarshah commented on v1/branch/branch.gunk:
```diff
@@ -139,39 +138,45 @@ type UpdateBranchRequest struct {
 	//                         Description: "Request executed successfully.",
+	//                         Schema: openapiv2.Schema{
+	//                                 JSONSchema: openapiv2.JSONSchema{
+	//                                         Ref: "#/definitions/branchBranch",
```
Should the Ref be "#/definitions/branch`es`Branch"?

mgfcmartinez replied:
similar to above: https://github.com/openbank/openbank/blob/b496bf680a51b141c09f9cb480b133d4d54aba6e/v1/branch/all.swagger.json#L294

ghazninattarshah commented on v1/branch/branch.gunk:
```diff
@@ -199,11 +204,16 @@ type BranchService interface {
 	//                         Description: "Request executed successfully.",
+	//                         Schema: openapiv2.Schema{
+	//                                 JSONSchema: openapiv2.JSONSchema{
+	//                                         Ref: "#/definitions/branchGetBranchesResponse",
```
Should the Ref be "#/definitions/branch`es`GetBranchesResponse?

mgfcmartinez replied:
see: https://github.com/openbank/openbank/pull/17#discussion_r314579504

ghazninattarshah commented on v1/branch/branch.gunk:
```diff
@@ -233,10 +243,15 @@ type BranchService interface {
 	//                         Description: "Branch created successfully.",
+	//                         Schema: openapiv2.Schema{
+	//                                 JSONSchema: openapiv2.JSONSchema{
+	//                                         Ref: "#/definitions/branchCreateBranchResponse",
```
Should the Ref be "#/definitions/branch`es`CreateBranchResponse?

mgfcmartinez replied:
see: https://github.com/openbank/openbank/pull/17#discussion_r314579504

ghazninattarshah commented on v1/profiles/profiles.gunk:
```diff
@@ -55,64 +54,69 @@ import (
 	//
 	// +gunk http.Match{
 	//         Method: "GET",
 	//         Path:   "/v1/profile",
```
path should be `v1/profiles`

ghazninattarshah commented on v1/profiles/profiles.gunk:
```diff
@@ -134,22 +138,22 @@ type ProfileService interface {
 	//
 	// +gunk http.Match{
 	//         Method: "GET",
 	//         Path:   "/v1/profile/card",
```
path should be `v1/profiles/card`

mgfcmartinez commented:
@ghazninattarshah fixed the account endpoint summaries and the profile endpoints, thanks!

nkristianto commented:
add @jwangsadinata  , since he has a better english :D

donskifarrell commented:
The comments LGTM 👍

mgfcmartinez commented:
@donskifarrell thanks! generated files have been added

jwangsadinata commented on v1/accounts/accounts.gunk:
```diff
@@ -78,13 +77,13 @@ type Account struct {
 	AmountDue types.Amount `pb:"6" json:"amount_due"`
 
-	// TODO: add comment.
+	// Available balance is the available balance of the account.
```
s/Available\ balance/AvailableBalance

mgfcmartinez replied:
nice catch, thanks!

jwangsadinata reviewed (approved):
one small comment, but LGTM. Thanks a lot @mgfcmartinez.

nkristianto reviewed (approved):
Looks good to me sir.

Commit:
accounts, atm, auth, branch, profiles, statements: cleanup

- Fix typos and grammatical errors, try to make the writing style more
  consistent

- Remove BasePath from swagger definitions to prevent it being doubled
  (e.g. /v1/v1/accounts/check)

- Fix missing/incorrect response schemas

- Make godoc comments more consistent

- Fix endpoint tags to ensure correct grouping
